src/blocks/plugins/dynamic-content/value/index.js

```diff
@@ -38,3 +51,68 @@ export const format = {
 };
 
 registerFormatType( name, format );
+
+const displayWaitlist = {};
+
+const displayData = ( element, value ) => {
+	const el = document.createElement( 'div' );
+	el.innerHTML = value;
+	value = el.textContent || el.innerText;
+
+	element.innerHTML = '<span>' + element.innerHTML + '</span>';
+	element.dataset.preview = value;
+};
+
+const withDynamicConditions = createHigherOrderComponent( BlockEdit => {
+	return props => {
+		if ( ! props.isSelected ) {
+			const id = `block-${ props.clientId }`;
+			const elements = document.querySelectorAll( `#${ id } o-dynamic` );
+
+			if ( ! elements?.length ) {
+				return <BlockEdit { ...props } />;
+			}
+
+			elements.forEach( element => {
+				const context = select( 'core/editor' ).getCurrentPostId();
+				const attrs = pick( Object.assign({ context }, element.dataset ), [ 'type', 'context', 'before', 'after', 'length', 'dateType', 'dateFormat', 'dateCustom', 'timeType', 'timeFormat', 'timeCustom', 'termType', 'termSeparator', 'metaKey' ]);
+
+				if ( 'postContent' === attrs.type ) {
+					return;
+				}
+
+				let value = select( 'themeisle-gutenberg/data' ).getDynamicData( attrs );
+				if ( undefined !== value ) {
+					displayData( element, value );
+				} else {
+					const attrsHash = hash( attrs );
+
+					if ( ! displayWaitlist[ attrsHash ]) {
+						displayWaitlist[ attrsHash ] = [ element ];
+
+						const interval = setInterval( () => {
+							value = select( 'themeisle-gutenberg/data' ).getDynamicData( attrs );
+
+							if ( undefined !== value ) {
+								clearInterval( interval );
+								displayWaitlist[ attrsHash ].forEach( el => displayData( el, value ) );
+							}
+						}, 5000 );
```

Comment:
I think we can put a limit on how many times this interval can run for a hash code.

Reply:
Also, you can delete the items after loading.

```javascript
displayWaitlist[ attrsHash ] = [];
```

No need to keep them in memory.